plugins/fcp_can_c/fcp_can_c/can_c_writer.py

```diff
@@ -366,10 +372,60 @@ def generate_device_sources(self) -> Generator[Tuple[str, str], None, None]:
         """
         for device_name, messages in self.device_messages.items():
             yield (
-                pascal_to_snake(device_name),
+                to_snake_case(device_name),
                 self.templates["device_can_c"].render(
-                    device_name_pascal=snake_to_pascal(device_name),
-                    device_name_snake=pascal_to_snake(device_name),
+                    device_name_pascal=to_pascal_case(device_name),
+                    device_name_snake=to_snake_case(device_name),
+                    messages=messages,
+                ),
+            )
+
+    def generate_rpc_headers(self) -> Generator[Tuple[str, str], None, None]:
+        """Generate C header files for devices with RPC.
+
+        Returns:
+            Generator: Tuple containing the device name and the file content.
+
+        """
+        for device in self.devices:
+            device_name = device.name
+            messages = self.device_messages.get(device_name, [])
+            rpc_get_id = device.rpc_get_id if device.rpc_get_id is not None else 0
+            rpc_ans_id = device.rpc_ans_id if device.rpc_ans_id is not None else 0
+
+            yield (
+                device_name,
+                self.templates["device_rpc_h"].render(
+                    device_name_pascal=to_pascal_case(device_name),
+                    device_name_snake=to_snake_case(device_name),
+                    messages=messages,
+                    rpc_get_id=rpc_get_id,
+                    rpc_ans_id=rpc_ans_id,
+                    services=device.services,
+                ),
+            )
+
+    def generate_rpc_sources(self) -> Generator[Tuple[str, str], None, None]:
+        """Generate C source files for devices with RPC.
+
+        Returns:
+            Generator: Tuple containing the device name and the file content.
+
+        """
+        for device in self.devices:
+            device_name = device.name
+            messages = self.device_messages.get(device_name, [])
+            rpc_get_id = device.rpc_get_id if device.rpc_get_id is not None else 0
+            rpc_ans_id = device.rpc_ans_id if device.rpc_ans_id is not None else 0
+
+            yield (
+                to_snake_case(device_name),
+                self.templates["device_rpc_c"].render(
+                    device_name_pascal=to_pascal_case(device_name),
+                    device_name_snake=to_snake_case(device_name),
                     messages=messages,
+                    rpc_get_id=rpc_get_id,
+                    rpc_ans_id=rpc_ans_id,
+                    services=device.services,
```

Comment:
this function is almost entirely copy paste from `generate_rpc_headers`, that must mean there's stuff we can factor out.